types/node/stream.d.ts

```diff
@@ -126,6 +126,16 @@ declare module 'stream' {
              * @since v18.0.0
              */
             destroyed: boolean;
+            /**
+             * Is true after 'close' has been emitted.
+             * @since v8.0.0
+             */
+            readonly closed: boolean;
+            /**
+             * Returns error if the stream has been destroyed with an error.
+             * @since v18.0.0
+             */
+            readonly errored: Error | null;
```

Comment:
IMO this should be undefined, so we get `Error` or nothing.
I understand `null` works with expression same way as undefined.

Reply:
@peterblazejewicz, based on the documentation, it should be `Error | null`. To confirm, I ran a test to check the `errored` value for every possible readable state except `errored`, and it's always `null`:

```
rusnak@Dawid-15 ~ % cat test.js
const { Readable } = require('node:stream');
const s = new Readable();
console.log(s.errored); // initial
s.push(null);
console.log(s.errored); // ended internally
s.read();
console.log(s.errored); // before readable ended tick
process.nextTick(() => {
  console.log(s.errored); // readable ended
});
rusnak@Dawid-15 ~ % node -v
v18.4.0
rusnak@Dawid-15 ~ % node test
null
null
null
null
```

Is there some reason to set its type to `undefined`, given that it will never be `undefined`?

Reply:
really sorry for this, I was wrong (confused by Node runtime version I've used). Checked with source, it's always null.

Reply:
No problem! Just thought for a moment, that there is some neat hack required, that would use `undefined` here 🙂